frontend/pages/login/[service].tsx

```diff
@@ -3,56 +3,85 @@ import { useTranslation } from "next-i18next"
 import { serverSideTranslations } from "next-i18next/serverSideTranslations"
 import { NextSeo } from "next-seo"
 import { useRouter } from "next/router"
-import { useEffect, useState } from "react"
+import { useCallback, useEffect, useState } from "react"
 import { toast } from "react-toastify"
 import { login } from "../../src/asyncs/login"
 import Spinner from "../../src/components/Spinner"
 import { useUserContext, useUserDispatch } from "../../src/context/user-info"
 import { fetchLoginProviders } from "../../src/fetchers"
+import { useAsync } from "../../src/hooks/useAsync"
 import { useLocalStorage } from "../../src/hooks/useLocalStorage"
 import { usePendingTransaction } from "../../src/hooks/usePendingTransaction"
 import { isInternalRedirect } from "../../src/utils/security"
 
-export default function AuthReturnPage({ services }) {
+export default function AuthReturnPage({ services }: { services: string[] }) {
   // Must access query params to POST to backend for oauth verification
   const { t } = useTranslation()
   const router = useRouter()
 
-  // Send only one request, prevent infinite loops
-  const [sent, setSent] = useState(false)
-
   const user = useUserContext()
   const dispatch = useUserDispatch()
 
-  const [pendingTransaction, _setPendingTransaction] = usePendingTransaction()
+  const [pendingTransaction] = usePendingTransaction()
   const [returnTo, setReturnTo] = useLocalStorage<string>("returnTo", "")
 
+  const { execute, status, error } = useAsync(
+    useCallback(() => login(dispatch, router.query), [dispatch, router.query]),
+    false,
+  )
+
+  useEffect(() => {
+    if (error) {
+      toast.error(t(error))
+    }
+  }, [t, error])
+
+  // Once router ready, perform login
+  useEffect(() => {
+    if (router.isReady) {
+      execute()
+    }
+  }, [router.isReady, execute])
+
+  // Redirect back to where user was, or userpage for unprompted login
```

Comment:
This comment seems to be unrelated to the next line now?

Reply:
I figure it pertains to the whole block below, with the state being an implementation detail of that section of code